### PR TITLE
[DR-2210] Update page styling

### PR DESCRIPTION
**Pull request by petesantos**
Ticket: https://broadworkbench.atlassian.net/browse/DR-2210

This change is mostly about updated the styling of the data explorer page to better match the mock:
https://projects.invisionapp.com/d/main#/console/21637590/457826296/inspect

I did make a change to the autocomplete input so it can accept an icon, in this case a search / magnifying glass icon.

## Before
image

## After
image

### CODE REVIEW

File: src/pages/library/common.js [hunk 1]
```diff
@@ -88,6 +90,7 @@ const Sidebar = ({ onSectionFilter, onTagFilter, sections, selectedSections, sel
 }
 
 const reverseText = _.flow(_.reverse, _.join(''))
+const contextualLength = 80
```
Comment (kyuksel) [inline src/pages/library/common.js]: Could this be inlined in the one place where it's used?

**Review by kyuksel — APPROVED** [review]
👍🏼  with one minor comment inline

File: src/components/input.js [hunk 2]
```diff
@@ -255,10 +255,14 @@ const withAutocomplete = WrappedComponent => ({
...
-        style: { width: '100%', display: 'inline-flex' }
+        style: { width: style?.width || '100%', display: 'inline-flex', position: 'relative' }
       }, [
+        inputIcon && icon(inputIcon, {
+          style: { transform: 'translateX(1.5rem)', alignSelf: 'center', color: colors.accent(), position: 'absolute', ...iconStyle },
+          size: 18
+        }),
         h(WrappedComponent, getInputProps({
-          style,
+          style: inputIcon ? { ...style, paddingLeft: '3rem' } : style,
```
Comment (petesantos) [inline src/components/input.js:265]: If an input icon is specified, shift the text over to allow space for the icon

File: src/pages/library/common.js (hunk 1 above)
Comment (petesantos) [reply src/pages/library/common.js]: Yes - I originally used it in multiple places.

File: src/pages/library/common.js [hunk 3]
```diff
@@ -183,24 +185,33 @@ export const SearchAndFilterComponent = ({ fullList, sidebarSections, customSort
...
-        div({ style: { display: 'flex', alignItems: 'center', height: '2.5rem' } }, [
-          div({ style: { flex: 1 } }),
-          h(Link, {
-            onClick: () => {
-              setSelectedSections([])
-              setSelectedTags([])
-            }
-          }, ['clear'])
-        ])
+    div({
```
Comment (petesantos) [inline src/pages/library/common.js:188]: Converted to grid, some whitespace changes follow

**Review by lu-c — APPROVED** [review]
Sorry! I thought i'd already approved this PR, and I didnt check its status again til now >.<

**Comment by kyuksel** [comment]
@lu-c does your team want to merge this PR now or wait until Pete comes back from vacation? Just checking :)

**Comment by lu-c** [comment]
Ah, Let's just merge it :)

**Pull request merged**